burstcamp/burstcamp/Scene/Tab/Detail/ViewModel/FeedDetailViewModel.swift

```diff
@@ -53,28 +58,111 @@ final class FeedDetailViewModel {
             feedDidUpdate: feedDidUpdate,
             openBlog: openBlog,
-            scrapButtonToggle: input.scrapButtonDidTap,
-            openActivityView: openActivityView
+            openActivityView: openActivityView,
+            scrapButtonToggle: scrapButtonToggle,
+            scrapButtonDisable: scrapButtonDisabled,
+            scrapButtonEnable: scrapButtonEnabled
         )
     }
 
+    private func updateFeed(uuid: String, state: Bool) {
+        Task {
+            do {
+                switch state {
+                case true: try await requestDeleteFeedScrapUser(uuid: uuid)
+                case false: try await requestCreateFeedScrapUser(uuid: uuid)
+                }
+            } catch {
+                self.dbUpdateSucceed.send(false)
+                return
+            }
+            self.dbUpdateSucceed.send(true)
+        }
+    }
+
     private func fetchFeed(uuid: String) {
         Task {
             // TODO: 오류 처리
-            let feedDict = try await requestFeed(uuid: uuid)
+            let feedDict = try await requestGetFeed(uuid: uuid)
+            print(feedDict.keys)
             let feedDTO = FeedDTO(data: feedDict)
-            let feedWriterDict = try await requestFeedWriter(uuid: feedDTO.writerUUID)
+            let feedWriterDict = try await requestGetFeedWriter(uuid: feedDTO.writerUUID)
             let feedWriter = FeedWriter(data: feedWriterDict)
             let feed = Feed(feedDTO: feedDTO, feedWriter: feedWriter)
 
             self.feed.send(feed)
         }
     }
 
-    private func requestFeed(uuid: String) async throws -> [String: Any] {
+    private func requestDeleteFeedScrapUser(uuid: String) async throws {
+        try await withCheckedThrowingContinuation({ continuation in
+            Firestore.firestore()
+                .collection("feed")
+                .document(uuid)
+                .collection("scrapUserUUIDs")
+                .document("userUUID")
+                .delete { error in
+                    if let error = error {
+                        continuation.resume(throwing: error)
+                        return
+                    }
+                    continuation.resume()
+                }
+        }) as Void
+    }
+
+    private func requestCreateFeedScrapUser(uuid: String) async throws {
+        try await withCheckedThrowingContinuation({ continuation in
+            Firestore.firestore()
+                .collection("feed")
+                .document(uuid)
+                .collection("scrapUserUUIDs")
+                .document("userUUID")
+                .setData([
+                    "userUUID": "userUUID",
+                    "scrapDate": Timestamp(date: Date())
+                ]) { error in
+                    if let error = error {
+                        continuation.resume(throwing: error)
+                        return
+                    }
+                    continuation.resume()
+                }
+        }) as Void
```

Comment:
Q.

what... is.... this..

Reply:
https://stackoverflow.com/questions/68706440/async-await-function-without-a-return-swift-5-5
ㅜㅜ